### ISSUE TITLE
CoMID generates incorrect triple index for `identity-triples` and `attest-key-triples`

### ISSUE THREAD

**Issue by amd-isaac** [veraison/corim#116]
When generating a CoMID with an entry in `identity-triples` (AKA `dev-identity-keys` in the `cocli` CoMID template), the resultant binary uses a `triples-map` index of `3` rather than the value of `2` in the [IETF draft spec](https://www.ietf.org/archive/id/draft-ietf-rats-corim-04.html#section-3.1.4-4.3.1). Similarly the `attest-key-triples` generates with an index of `2` when the spec says it should be `3`. Perhaps these two were swapped during the initial implementation? Or is the change in terminology intentional?

Code from `triples.go`: 
```
type Triples struct {
    ReferenceValues *[]ReferenceValue `cbor:"0,keyasint,omitempty" json:"reference-values,omitempty"`
    EndorsedValues  *[]EndorsedValue  `cbor:"1,keyasint,omitempty" json:"endorsed-values,omitempty"`
    AttestVerifKeys *[]AttestVerifKey `cbor:"2,keyasint,omitempty" json:"attester-verification-keys,omitempty"`
    DevIdentityKeys *[]DevIdentityKey `cbor:"3,keyasint,omitempty" json:"dev-identity-keys,omitempty"`

    Extensions
}
```

Minimal example `template.json` file:
```
{
  "tag-identity": {
    "id": "eedf041b-2dbe-4f1a-968f-cc64f21db8ae"
  },
  "entities": [
    {
      "name": "Tag Creator Jr.",
      "roles": [
        "tagCreator"
      ]
    }
  ],
  "triples": {
    "dev-identity-keys": [
      {
        "environment": {
          "class": {
            "vendor": "Vendor Esq."
          }
        },
        "verification-keys": [
          {
            "type": "pkix-base64-key",
            "value": "-----BEGIN PUBLIC KEY-----\nMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQD2mhCBSTYrACSHdLtbl3McOoPm\nZYQX40XjPbzKFsnc/5lKxjjVb3EAVbOm9MXOI+0PS/6iKhwTKox4d7LoY4qHvIZ1\nZka2tYxUxRqXLzzcWlUvopuH/O5S7kgQc4IU7+3KZ1dnacsHcEzBpO4uhapjCn7z\nUdWI3g/M4Xz8EQGRlQIDAQAB\n-----END PUBLIC KEY-----"
          }
        ]
      }
    ]
  }
}
```

Output generated with v1.1.2: 
```
00000000: a301 a100 50ee df04 1b2d be4f 1a96 8fcc  ....P....-.O....
00000010: 64f2 1db8 ae02 81a2 006f 5461 6720 4372  d........oTag Cr
00000020: 6561 746f 7220 4a72 2e02 8100 04a1 0381  eator Jr........
00000030: 82a1 00a1 016b 5665 6e64 6f72 2045 7371  .....kVendor Esq
00000040: 2e81 d902 2a79 010f 2d2d 2d2d 2d42 4547  ....*y..-----BEG
00000050: 494e 2050 5542 4c49 4320 4b45 592d 2d2d  IN PUBLIC KEY---
00000060: 2d2d 0a4d 4947 664d 4130 4743 5371 4753  --.MIGfMA0GCSqGS
00000070: 4962 3344 5145 4241 5155 4141 3447 4e41  Ib3DQEBAQUAA4GNA
00000080: 4443 4269 514b 4267 5144 326d 6843 4253  DCBiQKBgQD2mhCBS
00000090: 5459 7241 4353 4864 4c74 626c 334d 634f  TYrACSHdLtbl3McO
000000a0: 6f50 6d0a 5a59 5158 3430 586a 5062 7a4b  oPm.ZYQX40XjPbzK
000000b0: 4673 6e63 2f35 6c4b 786a 6a56 6233 4541  Fsnc/5lKxjjVb3EA
000000c0: 5662 4f6d 394d 584f 492b 3050 532f 3669  VbOm9MXOI+0PS/6i
000000d0: 4b68 7754 4b6f 7834 6437 4c6f 5934 7148  KhwTKox4d7LoY4qH
000000e0: 7649 5a31 0a5a 6b61 3274 5978 5578 5271  vIZ1.Zka2tYxUxRq
000000f0: 584c 7a7a 6357 6c55 766f 7075 482f 4f35  XLzzcWlUvopuH/O5
00000100: 5337 6b67 5163 3449 5537 2b33 4b5a 3164  S7kgQc4IU7+3KZ1d
00000110: 6e61 6373 4863 457a 4270 4f34 7568 6170  nacsHcEzBpO4uhap
00000120: 6a43 6e37 7a0a 5564 5749 3367 2f4d 3458  jCn7z.UdWI3g/M4X
00000130: 7a38 4551 4752 6c51 4944 4151 4142 0a2d  z8EQGRlQIDAQAB.-
00000140: 2d2d 2d2d 454e 4420 5055 424c 4943 204b  ----END PUBLIC K
00000150: 4559 2d2d 2d2d 2d                        EY-----
```

The byte located at `0x2e` is set to `0x03` when it should be `0x02`.

**Comment by yogeshbdeshpande** [comment]
@amd-isaac : Thank you so much for pointing this valid bug.  Sergei has done the fix, will be merged by next week.

Any other feedback (like this) is also very much appreciated..

Is it possible for you to point to your open source repository (if available), where you are using CoRIM/CoMID data model.

We just want to understand better how this package of work is been used with a pure intention of improving this even more for our CoRIM Community..

Thanks again for pointing the issue.

**Comment by thomas-fossati** [comment]
Fixed in #118
